metalearners/xlearner.py

```diff
@@ -99,31 +99,39 @@ def fit_all_nuisance(
 
         qualified_fit_params = self._qualified_fit_params(fit_params)
 
-        self._cvs: list = []
+        # TODO: Move this to object initialization.
+        if not synchronize_cross_fitting:
+            raise ValueError(
+                "The X-Learner does not support synchronize_cross_fitting=False."
+            )
+
+        self._cv_split_indices = self._split(X)
+        self._treatment_cv_split_indices = {}
 
         for treatment_variant in range(self.n_variants):
             self._treatment_variants_indices.append(w == treatment_variant)
-            if synchronize_cross_fitting:
-                cv_split_indices = self._split(
-                    index_matrix(X, self._treatment_variants_indices[treatment_variant])
+            treatment_indices = np.where(
+                self._treatment_variants_indices[treatment_variant]
+            )[0]
+            self._treatment_cv_split_indices[treatment_variant] = [
+                (
+                    np.intersect1d(train_indices, treatment_indices),
+                    np.intersect1d(test_indices, treatment_indices),
                 )
-            else:
-                cv_split_indices = None
-            self._cvs.append(cv_split_indices)
+                for train_indices, test_indices in self._cv_split_indices
```

Comment:
I am not sure I understand this with `_cv_split_indices` correctly. Isn't `_cv_split_indices` the list of lists of each of the folds, i. e [[indices fold_1], [indices fold_2], ...[indices fold_n]]? The code here seems to assume a form
[[train_indices_fold_1, test_indices_fold_1], ...., [train_indices_fold_n, test_indices_fold_n]].

Reply:
`cv_split_indices` is a list of tuples. Its length is equal to the number of folds. Every tuple contains two elements, one array with all indices used for training, one array with all indices used for testing. 
The union of both arrays should always give the entire dataset.

Does that answer your question?

Reply:
yep, thanks!

Reply:
I added type hints to hopefully make this a little more transparent: [b005eb7](https://github.com/Quantco/metalearners/pull/84/commits/b005eb70b9e9f1991028dd75c0b708c010654c77)

The type alias definition is here:

https://github.com/Quantco/metalearners/blob/fccde2cd20d508a948beab7f3495aaae84ba8420/metalearners/_typing.py#L41-L45